GitUI/UserControls/RevisionGrid/ChangeCount.cs

```diff
@@ -0,0 +1,91 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Text;
+using GitCommands;
+
+namespace GitUI
+{
+    public class ChangeCount
+    {
+        // Count for artificial commits
+        public IReadOnlyList<GitItemStatus> Changed { get; set; }
+        public IReadOnlyList<GitItemStatus> New { get; set; }
+        public IReadOnlyList<GitItemStatus> Deleted { get; set; }
+        public IReadOnlyList<GitItemStatus> SubmodulesChanged { get; set; }
+        public IReadOnlyList<GitItemStatus> SubmodulesDirty { get; set; }
+
+        public bool HasChanges
+            => (Changed?.Count ?? 0) > 0
+               || (New?.Count ?? 0) > 0
+               || (Deleted?.Count ?? 0) > 0
+               || (SubmodulesChanged?.Count ?? 0) > 0
+               || (SubmodulesDirty?.Count ?? 0) > 0;
```

Comment:
public bool HasChanges
    => Changed is not null
       || New is not null
       || Deleted is not null
       || SubmodulesChanged is not null
       || SubmodulesDirty is not null;

Reply:
This will require that C#9 is enabled for the project.
Also such changes should be globally enabled.
I will submit another PR for this change.

Reply:
The resulting code with the Count checks is not better in my opinion, so I suggest this is not changed:
```C#
        public bool HasChanges
            => (Changed is not null && Changed.Count > 0)
               || (New is not null && New.Count > 0)
               || (Deleted is not null && Deleted.Count > 0)
               || (SubmodulesChanged is not null && SubmodulesChanged.Count > 0)
               || (SubmodulesDirty is not null && SubmodulesDirty.Count > 0);
```